contracts/colonyNetwork/ColonyNetworkMining.sol

```diff
@@ -173,7 +174,11 @@ contract ColonyNetworkMining is ColonyNetworkStorage {
     }
 
     // II. Disburse reputation and tokens
-    IMetaColony(metaColony).mintTokensForColonyNetwork(realReward);
+    uint256 chainId = getChainId();
+    // On Xdai, we can only use bridged tokens, so no minting
+    if (chainId != 100 && chainId != 265669100) {
```

Comment:
One idea is to include these types of conditionals in the `GetChainId` contract itself, so these contracts can be more semantic, e.g. `if isMainnet()` etc.